duckduckgo/Android
# Improve Autofill dialogs

Task/Issue URL: https://app.asana.com/0/488551667048375/1202650338354716/f

### Description
First pass at improving the autofill dialogs UX. Includes changes to:

- saving a credential
- selecting a credential
- updating a credential
- support for dark theme
- showing initial onboarding text when offering to save first login


Does not include:

- ensuring landscape mode is functional
- update username


### Steps to test this PR

1. Test entering login credentials for the first time; verify the dialog includes onboarding explanation about where credentials are store. If you decline and try again, verify you see the explanation again.
2. Accept to save credentials, then try to save another set of credentials. Verify you do not see the onboarding explanation any more
3. Test selecting login credentials when you have 
    - none saved (verify no dialog)
    - 1 saved (verify dialog has only 1 primary CTA with login)
    - 2 saved (verify dialog has 1 primary and 1 secondary button)
    - 3 saved (verify dialog has 1 primary and 2 secondary buttons)
    - 4 saved (verify dialog has 1 primary, 1 secondary login and 1 secondary button for "more options". Verify clicking on more options reveals all the remaining ones)
4.  Test updating a password; verify you see a password preview of the correct length for the new password
5. Test dark theme looks reasonable


### UI changes
![Screenshot_20220722_103128](https://user-images.githubusercontent.com/1336281/180410689-4496d339-d317-4ffc-8ec4-79c5174a14a6.png)
![Screenshot_20220722_103340](https://user-images.githubusercontent.com/1336281/180410744-3f0fcffe-1f1b-4f4f-8f4c-741b6725c870.png)
![Screenshot_20220722_103223](https://user-images.githubusercontent.com/1336281/180410836-6a91c9af-22cb-4129-b389-8972a74f81b6.png)
![Screenshot_20220722_103220](https://user-images.githubusercontent.com/1336281/180410844-ba128481-38d6-42d4-b6bc-1d3172279702.png)
![Screenshot_20220722_103208](https://user-images.githubusercontent.com/1336281/180410852-4679325a-76cc-4c17-86d5-c2c587c097c2.png)
![Screenshot_20220722_103157](https://user-images.githubusercontent.com/1336281/180410857-8d09af03-4f8c-4d82-ae8a-b97f634a38d1.png)
![Screenshot_20220722_103139](https://user-images.githubusercontent.com/1336281/180410863-c8918a4a-1ad8-4e96-9cf4-959cb626bdb6.png)
![Screenshot_20220722_103240](https://user-images.githubusercontent.com/1336281/180410889-fbdb6794-8ceb-4dec-a961-cbd59396266d.png)
![Screenshot_20220722_103503](https://user-images.githubusercontent.com/1336281/180410961-1aebebe3-56d7-4f54-a0d6-e5977f365a2c.png)
